lib/msf/core/modules/loader/executable.rb

```diff
@@ -11,13 +11,38 @@ def loadable?(path)
     File.directory?(path)
   end
 
-  def loadable_module?(parent_path, type, module_reference_name)
-    full_path = module_path(parent_path, type, module_reference_name)
+  # @param [String] parent_path Root directory to load modules from
+  # @param [String] type Such as auxiliary, exploit, etc
+  # @param [String] module_reference_name The module reference name, without the type prefix
+  # @param [nil,Msf::Modules::Metadata::Obj] cached_metadata
+  # @return [Boolean] True this loader can load the module, false otherwise
+  def loadable_module?(parent_path, type, module_reference_name, cached_metadata: nil)
+    full_path = cached_metadata&.path || module_path(parent_path, type, module_reference_name)
     script_path?(full_path)
   end
 
   protected
 
+  def read_script_runtime(full_path)
+    # Extract the runtime from the first line of the script, i.e.
+    #   #!/usr/bin/env python
+    #   //usr/bin/env go run "$0" "$@"; exit "$?"
+    first_line = File.open(full_path) { |f| f.readline }
+    first_line[%r{(?:#!|\/)\/usr\/bin\/env\s+(\w+)}, 1]
```

Comment:
```suggestion
    first_line[%r{^(?:#!|\/)\/usr\/bin\/env\s+(\w+)}, 1]
```

Are you sure `//` is a valid construct for a shebang? Also, thanks to the `%r` specifier, I'm not sure `/` needs to be escaped, do they?

Reply:
`//` is for the go modules: https://github.com/rapid7/metasploit-framework/blob/fc7688cc56d7f4cc1508fef73b9feb3dca473a07/modules/auxiliary/scanner/msmail/host_id.go#L1 :+1:

Reply:
Golfed and tests added :+1: